test/unit/BPool.t.sol

```diff
@@ -23,87 +39,130 @@ abstract contract Base is Test, BConst {
       tokens[i] = makeAddr(i.toString());
     }
   }
+
+  function _setValues(FuzzScenario memory _fuzz) internal {
+    // Create mocks
+    for (uint256 i = 0; i < tokens.length; i++) {
+      vm.mockCall(tokens[i], abi.encodeWithSelector(IERC20(tokens[i]).transfer.selector), abi.encode(true));
+      vm.mockCall(tokens[i], abi.encodeWithSelector(IERC20(tokens[i]).transferFrom.selector), abi.encode(true));
+    }
```

Comment:
internalize each! 🙏 
so, it should look sth like:
```solidity
// within PoolJoinTest
function _setValues(FuzzScenario memory _fuzz) internal {
    for (i...) {
        _setTokens(tokens[i])
        _mockTokenTransfers(tokens[i]) // hardcoded
        _setTokenBalances(tokens[i], _fuzz.balance[i]); // fuzzed
    }
    _setPublicSwap(true); // hardcoded
    _setFinalize(true); // hardcoded
    _setTotalSupply(_fuzz.totalSupply);
}
```

the setters are common, reusable for ALL test contracts, the happy paths and scenarios are specific to each method

Reply:
addressed in #11 😉